src/eth_rpc/config.rs

```diff
@@ -24,4 +24,19 @@ impl RPCConfig {
         }
         Ok(config)
     }
+
+    pub fn new_test_config() -> Self {
+        // Hardcode the socket address for testing environment
+        Self::new("127.0.0.1:3030".to_string())
+    }
+
+    pub fn new_test_config_from_port(port: u16) -> Self {
+        let mut config = Self::new_test_config();
+        // Remove port from socket address and replace it with provided port
+        let parts: Vec<&str> = config.socket_addr.split(':').collect();
+        if let Some(addr) = parts.first() {
+            config.socket_addr = format!("{}:{}", addr, port);
+        }
+        config
+    }
```

Comment:
Can we just put these behind a `#[cfg(feature = "testing")]`?

Reply:
good idea, just added